### PR TITLE
Third milestone

### PR DESCRIPTION
**Pull request by sumancrest0001**
Features  of the milestone project.
1. User authentication with devise gem.
2. Creating posts and association with the user.
3. Unit tests for the models and controllers.

### CODE REVIEW

**Comment by noomdalv** [comment]
Corrections applied:

- [ ] 1. Guests can only see the sign in page
- [ ] 2. users#index should now display all the users
- [ ] 3. users can now create posts correctly
- [ ] 4. posts are displayed with their author, content and creation date
- [ ] 5. Sign up form  now allow users to sign up correctly with their name, email and password attribs.

File: config/routes.rb [hunk 1]
```diff
@@ -1,3 +1,8 @@
 Rails.application.routes.draw do
-  # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
+  get 'users/index'
```
Comment (mcrd25) [inline config/routes.rb]: [OPTIONAL]
You should consider that it is a common convention for your `users#index` or any index path to be like `users` as opposed to `users/index`.

**Comment by sumancrest0001** [comment]
Corrected the redirection issue after signout. Thank you for the optional change. We are planning to make a profile page in milestone-4 as there was no clear point for profile page in milestone-3.

**Comment by sumancrest0001** [comment]
Showing users without login is fixed.

File: app/controllers/posts_controller.rb [hunk 2]
```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+class PostsController < ApplicationController
+  before_action :logged_in?, only: %i[show]
```
Comment (mosaaleb) [inline app/controllers/posts_controller.rb]: - [ ] **[OPTIONAL]** There is already helper method for devise that require login [`authenticate_user!`](https://github.com/plataformatec/devise#controller-filters-and-helpers)

**Pull request merged**